### DISCUSSION TITLE
Battlenet Provider adding scope (SvelteKit)

### DISCUSSION THREAD

**Discussion by JoschiZ** [nextauthjs/next-auth#6220] [Help] [answered]
Hey guys,

I can't seem to find how to set a scope for the battlenet provider.
Other provider like `Apple.js` have something like the following:
```js
        authorization: {
            params: { scope: "name email", response_mode: "form_post" },
        },
```
But the Battlenet Provider doesn't have this field and just adding it in the options like this doesn't seem to work for me:
```js
export const handle = SvelteKitAuth({
  providers:
    [GitHub({ clientId: GITHUB_ID, clientSecret: GITHUB_SECRET }),
    BNet({
      clientId: BNET_TEST_ID,
      clientSecret: BNET_TEST_SECRET,
      issuer: "https://eu.battle.net/oauth",
      authorization: {
        params: {
          scope: "wow.profile"
        }
      }
    })],

  callbacks: {
    async jwt({ token, user, account, profile, isNewUser }) {
      if (account) {
        console.log(account)
      }
    },
  },
})
```
`account.scope` only includes `openid`. And looking at the network the GET header has the scope listed as `"openid profile email"`. Profile and email aren't even scopes according to battlenets documentation.

I'm quite new in js/webdev so I might be missing something obvious here and am not sure where to start looking myself.

**Comment by 1337cookie** [comment, marked as answer]
Adding ```scope: "wow.profile"``` will give the access token you receive permission to query the users WoW profile. It should appear on the blizzard auth flow page that the app is requesting access to the WoW Profile information. 
Edit: I tested the same implementation you have and the following does ask for profile access:
```js
BattleNet({ clientId: process.env.BATTLENET_CLIENT_ID, clientSecret: process.env.BATTLENET_CLIENT_SECRET, issuer: 'https://us.battle.net/oauth', authorization: {params: { 'scope': 'wow.profile'}}}),
```
Using that access token you can then query the WoW profile API documented here: 
https://develop.battle.net/documentation/world-of-warcraft/profile-apis

The docs are a bit cumbersome. But here is the part about scopes:
https://develop.battle.net/documentation/guides/using-oauth#:~:text=Scopes%20and%20OAuth%20enabled%20APIs

**Reply by JoschiZ** [reply]
# Edit:
This seems to be fixed now. At least it works after updating the packages.
# 

Hey,
thanks for the reply.
I think I got the conceptual hang now of what scopes do, but there seems to be a SvelteKit (@auth/core) specific problem with at least the battlenet provider.
Comparing the @auth/core/provider/battlenet.js and the equivalent over in next-auth the only difference is the well-known link being present in next and not in core. I tried adding that as a shot in the dark and it did not solve the problem.

In this SvleteKit implementation the GitHub scope request for workflows works without any problems.
The Battlenet request for the WoW Data does not. Neither does it ask, nor is the access registered in Bnets connected app overview.
(`hooks.server.ts` from your SvelteKit template)
```ts
export const handle = SvelteKitAuth({
  providers:
    [BNet({
      clientId: BNET_TEST_ID,
      clientSecret: BNET_TEST_SECRET,
      issuer: 'https://eu.battle.net/oauth',
      authorization: { params: { 'scope': 'wow.profile' } }
    }),
    GitHub({
      clientId: GITHUB_ID,
      clientSecret: GITHUB_SECRET,
      authorization: { params: { "scope": "workflow" } },
    })
    ],

  callbacks: {
    async jwt({ token, user, account, profile, isNewUser }) {
      if (account) {
        console.log(account)
      }
    },
  },
})
```

I also loaded up your Next.js template and it works without any problems over there.

```ts
export const authOptions: NextAuthOptions = {
  // https://next-auth.js.org/configuration/providers/oauth
  providers: [
    GithubProvider({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET,
      authorization: {params: {"scope":"workflow"}}
    }),
    BattleNetProvider({
      clientId: process.env.BNET_TEST_ID,
      clientSecret: process.env.BNET_TEST_SECRET,
      issuer: "https://eu.battle.net/oauth",
      authorization: {params: { 'scope': 'wow.profile'}}
    })
  ],
  theme: {
    colorScheme: "light",
  },
  callbacks: {
    async jwt({ token }) {
      token.userRole = "admin"
      return token
    },
  },
}

```
